src/classes/CountertopWorker.js

```diff
@@ -0,0 +1,132 @@
+import { IAppliance } from '@tvkitchen/base-interfaces'
+import { Payload } from '@tvkitchen/base-classes'
+import { applianceEvents } from '@tvkitchen/base-constants'
+import { consoleLogger } from '../tools/loggers'
+import kafka from '../tools/kafka'
+import { getStreamTopic } from '../tools/utils/countertop'
+
+/**
+ * A CountertopWorker (aka Line Cook) is responsible for monitoring specific data streams
+ * and processing those streams using TV Kitchen Appliances to create new data.
+ *
+ * CountertopWorker instances consume Payloads from Kafka and pass them to their Appliances.
+ * CountertopWorker instances listen for emitted Payloads from appliances and process them.
+ */
+class CountertopWorker {
+	logger = null
+
+	consumer = null
+
+	producer = null
+
+	admin = null
+
+	appliance = null
+
+	stream = null
+
+	/**
+	 * Create a new CountertopWorker.
+	 *
+	 * @param  {Class}            Appliance     The class for the appliance that this worker will use.
+	 * @param  {CountertopStream} stream        A stream that ends in this worker's station.
+	 * @param  {Object}           configuration Configuration settings for the Appliance.
+	 */
+	constructor(
+		Appliance,
+		applianceSettings = {},
+		{
+			stream,
+			logger = consoleLogger,
+		} = {},
+	) {
+		this.stream = stream
+		this.logger = logger
+		this.consumer = kafka.consumer({ groupId: stream.id })
```

Comment:
Should we give workers their own UUIDs (rather than relying on the stream UUID)

There should only ever be one worker per stream, so it would be 1:1, but still.

Reply:
Seems like a good idea. If the constraint you describe (and which I still don't fully understand) holds it's unneeded, but less coupling seems good.